### PR TITLE
Implement explicit HTTP "basic" authentication support

### LINKED ISSUE #599: Accessing HTTP(s) Keyword with Basic Auth
**Is it a duplicate question?**
No, didn't find anything similar

Attempting to access a URL using basic auth in the URL
https://user:password@HOSTNAME

It appears that the page is not loading.  This works in the browser.  Is it even possible to access a URL using basic auth in the URL or is there a different way to format the URL?

**Info**
Uptime Kuma Version: 1.7.3
Using Docker?: Yes
Docker Version:  20.10.7
Node.js Version (Without Docker only):
OS: Ubuntu 21.04
Browser: Chrome / Brave

### PR DESCRIPTION
**Pull request by ivanbratovic**
# Description

Fixes #599, #768, #834.

Users seem to have trouble with configuring HTTP "basic" auth. This PR adds new inputs for HTTP(s) and Keyword monitors to set HTTP "basic" auth username and password. These are encoded in Base64 and added to the headers of the request.

This makes adding endpoints with HTTP "basic" authentication a bit more user-friendly, not requiring manual encoding and adding to the Headers section. However, the old way still works and "basic" auth monitors created before the upgrade should continue working normally.

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and test it
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- ~My code needed automated testing. I have added them (this is optional task)~

## Screenshots

image

### CODE REVIEW

**Comment by Saibamen** [comment]
You can add screenshots directly to GitHub. Just paste it to comment field

image

File: server/model/monitor.js [hunk 1]
```diff
@@ -150,6 +169,7 @@ class Monitor extends BeanModel {
                             "Accept": "*/*",
                             "User-Agent": "Uptime-Kuma/" + version,
                             ...(this.headers ? JSON.parse(this.headers) : {}),
+                            ...(basicauthHeader)
```
Comment (Saibamen) [inline server/model/monitor.js:172]:
```suggestion
                            ...(basicauthHeader),
```

File: server/model/monitor.js [hunk 2]
```diff
@@ -141,6 +152,14 @@ class Monitor extends BeanModel {
                     // Do not do any queries/high loading things before the "bean.ping"
                     let startTime = dayjs().valueOf();
 
+                    // HTTP basic auth
+                    let basicauthHeader = {};
+                    if (this.basicauth_user) {
+                        basicauthHeader = {
+                            "Authorization": "Basic " + this.encodeB64(this.basicauth_user, this.basicauth_pass)
```
Comment (Saibamen) [inline server/model/monitor.js:159]:
```suggestion
                            "Authorization": "Basic " + this.encodeB64(this.basicauth_user, this.basicauth_pass),
```

File: server/model/monitor.js [hunk 3]
```diff
@@ -141,6 +152,14 @@ class Monitor extends BeanModel {
                     // Do not do any queries/high loading things before the "bean.ping"
                     let startTime = dayjs().valueOf();
 
+                    // HTTP basic auth
+                    let basicauthHeader = {};
```
Comment (Saibamen) [inline server/model/monitor.js:156]:
```suggestion
                    let basicAuthHeader = {};
```

File: src/pages/EditMonitor.vue [hunk 4]
```diff
@@ -265,6 +265,15 @@
...
                                     <textarea id="headers" v-model="monitor.headers" class="form-control" :placeholder="headersPlaceholder"></textarea>
                                 </div>
+
+                                <!-- HTTP Basic Auth -->
+                                <h4 class="mt-5 mb-2">{{ $t("HTTP Basic Auth") }}</h4>
+                                <div class="my-3">
+                                    <label for="basicauth" class="form-label">{{ $t("Username") }}</label>
+                                    <input id="basicauth-user" v-model="monitor.basicauth_user" type="text" class="form-control" :placeholder="Username">
+                                    <label for="basicauth" class="form-label">{{ $t("Password") }}</label>
+                                    <input id="basicauth-pass" v-model="monitor.basicauth_pass" type="password" class="form-control" :placeholder="Password">
```
Comment (Saibamen) [inline src/pages/EditMonitor.vue:275]: Untranslated placeholders

**Comment by ivanbratovic** [comment]
> You can add screenshots directly to GitHub. Just paste it to comment field
Thanks for the tip! I've edited the original text.

Also, apreciate your time for the review. I agree with the changes so I'll implement these ASAP.

**Comment by ivanbratovic** [comment]
Rebased with upstream and implemented @Saibamen's suggestions.

**Review by Saibamen — APPROVED** [review]

**Review by deefdragon — APPROVED** [review]

**Pull request merged**